cisc/certificate.go

```diff
@@ -0,0 +1,329 @@
+GkxldCdzIEVuY3J5cHQgQXV0aG9yaXR5IFgzMIIBIjANBgkqhkiG9w0BAQEFAAOC
+AQ8AMIIBCgKCAQEAnNMM8FrlLke3cl03g7NoYzDq1zUmGSXhvb418XCSL7e4S0EF
+q6meNQhY7LEqxGiHC6PjdeTm86dicbp5gWAf15Gan/PQeGdxyGkOlZHP/uaZ6WA8
+SMx+yk13EiSdRxta67nsHjcAHJyse6cF6s5K671B5TaYucv9bTyWaN8jKkKQDIZ0
+Z8h/pZq4UmEUEz9l6YKHy9v6Dlb2honzhT+Xhq+w3Brvaw2VFn3EK6BlspkENnWA
+a6xK8xuQSXgvopZPKiAlKQTGdMDQMc2PMTiVFrqoM7hD8bEfwzB/onkxEz0tNvjj
+/PIzark5McWvxI0NHWQWM6r6hCm21AvA2H3DkwIDAQABo4IBfTCCAXkwEgYDVR0T
+AQH/BAgwBgEB/wIBADAOBgNVHQ8BAf8EBAMCAYYwfwYIKwYBBQUHAQEEczBxMDIG
+CCsGAQUFBzABhiZodHRwOi8vaXNyZy50cnVzdGlkLm9jc3AuaWRlbnRydXN0LmNv
+bTA7BggrBgEFBQcwAoYvaHR0cDovL2FwcHMuaWRlbnRydXN0LmNvbS9yb290cy9k
+c3Ryb290Y2F4My5wN2MwHwYDVR0jBBgwFoAUxKexpHsscfrb4UuQdf/EFWCFiRAw
+VAYDVR0gBE0wSzAIBgZngQwBAgEwPwYLKwYBBAGC3xMBAQEwMDAuBggrBgEFBQcC
+ARYiaHR0cDovL2Nwcy5yb290LXgxLmxldHNlbmNyeXB0Lm9yZzA8BgNVHR8ENTAz
+MDGgL6AthitodHRwOi8vY3JsLmlkZW50cnVzdC5jb20vRFNUUk9PVENBWDNDUkwu
+Y3JsMB0GA1UdDgQWBBSoSmpjBH3duubRObemRWXv86jsoTANBgkqhkiG9w0BAQsF
+AAOCAQEA3TPXEfNjWDjdGBX7CVW+dla5cEilaUcne8IkCJLxWh9KEik3JHRRHGJo
+uM2VcGfl96S8TihRzZvoroed6ti6WqEBmtzw3Wodatg+VyOeph4EYpr/1wXKtx8/
+wApIvJSwtmVi4MFU5aMqrSDE6ea73Mj2tcMyo5jMd6jmeWUHK8so/joWUoHOUgwu
+X4Po1QYz+3dszkDqMp4fklxBwXRsW10KXzPMTZ+sOPAveyxindmjkW8lGy+QsRlG
+PfZ+G6Z6h7mjem0Y+iWlkYcV4PIWL1iwBi8saCbGS5jN2p8M+X+Q7UNKEkROb3N6
+KOqkqm57TH2H3eDJAkSnh6/DNFu0Qg==
+-----END CERTIFICATE-----`
+
+// LetsEncryptURL is the URL needed to create a new client
+const LetsEncryptURL = "https://acme-v01.api.letsencrypt.org/directory"
+
+// Convert a certificate of type string into certificate of type x509.Certificate
+func pemToCertificate(certPem string) (*x509.Certificate, error) {
+	block, _ := pem.Decode([]byte(certPem))
+	if block == nil {
+		return nil, errors.New("Fail to decode pem")
+	}
+
+	return x509.ParseCertificate(block.Bytes)
+}
+
+// Verify if the string given as arguments correspond to a certificate
+// Return true if it is the case and false otherwise
+func isCert(cert string) bool {
+	_, err := pemToCertificate(cert)
+	return err == nil
+}
+
+// Check if the string in argument correspond to a valid certificate
+// Return true if it is the case and false otherwise
+func isValid(cert string) bool {
+	rootPool := x509.NewCertPool()
+	certRoots, err := pemToCertificate(LetsencryptCert)
+	if err != nil {
+		log.Info("The root certificate is not valid")
+		return false
+	}
+	rootPool.AddCert(certRoots)
+	certCheck, err := pemToCertificate(cert)
+	if err != nil {
+		log.Info("The checked certificate is not valid")
+		return false
+	}
+	opts := x509.VerifyOptions{
+		DNSName: certCheck.DNSNames[0],
+		Roots:   rootPool,
+	}
+	if _, err := certCheck.Verify(opts); err != nil {
+		log.Info("The certificate is not valid: " + err.Error())
+		return false
+	}
+	log.Info("Certificate OK")
+	return true
+}
+
+// Renew the certificate that is caracterized by its string version
+// and returns the new certificate
+func renewCert(cert string) (string, error) {
```

Comment:
I feel it's better if `renewCert` and `isValid` take `cert []byte` rather than string. This will save you some string/[]byte conversion.